### ISSUE TITLE
eslint silently ignores missing files

### ISSUE THREAD

**Issue by timbertson** [eslint/eslint#7390]
- **ESLint Version:** 3.8.1
- **Node Version:** 6.3.1
- **npm Version:** 3.10.3

**Please show your full configuration:**

none (default)

**What did you do? Please include the actual source code causing the issue.**

``` sh
$ eslint missing-file
```

**What did you expect to happen?**

A nonzero exit status, and an error. It sounds like this used to happen:
- https://github.com/eslint/eslint/issues/1296
- https://github.com/eslint/eslint/issues/1831

**What actually happened? Please include the actual, raw output from ESLint.**

No output, 0 exit status.

**Comment by not-an-aardvark** [comment]
Thanks for the report. I can confirm that ESLint does this, but I'm not sure whether it's intended behavior.

**Comment by timbertson** [comment]
From the issues I linked there was discussion about _how_ it should fail, but there did at least seem to be agreement that it _should_ fail (and that it did at the time).

**Comment by mysticatea** [comment]
I agree that it should fail.

**Comment by not-an-aardvark** [comment]
**TSC Summary**: This issue proposes changing the behavior of ESLint to report a fatal error if nonexistent files/globs are passed on the command line. This would be a breaking change. The TSC [discussed this](https://github.com/eslint/tsc-meetings/blob/b76a2f0ba1f8701602e7d5178f28b3a80bf98a27/notes/2017/2017-03-30.md#breaking-report-file-not-found-errors) last year to consider adding it to the 4.0 release, but the resolution at the time was to defer it and consider changing the behavior in 5.0.

**TSC Question**: Should we accept this change for 5.0?

**Comment by not-an-aardvark** [comment]
This issue was accepted in [today's TSC meeting](https://github.com/eslint/tsc-meetings/blob/db3c0010144ba403a95daf3611af24ee271b3c78/notes/2018/2018-03-01.md#eslint-silently-ignores-missing-files).
